### PR TITLE
plugin/log: remove OutputFile

### LINKED ISSUE #1216: plugin/log: OutputFile in Rule struct is not used
`OutputFile` in `Rule` struct from log.go is not used and can be removed.

It can cause some confusion that an output file can be specified but the log plugin only logs to stdout anyway.

### PR DESCRIPTION
**Pull request by miekg**
We use stdout for everything, remove OutputFile as it isn't used and
causes confusion.

Fixes #1216

### CODE REVIEW

File: plugin/log/setup.go [hunk 1]
```diff
@@ -57,19 +51,20 @@ func logParse(c *caddy.Controller) ([]Rule, error) {
...
-				OutputFile: DefaultLogFilename,
-				Format:     DefaultLogFormat,
+				NameScope: ".",
+				Format:    DefaultLogFormat,
 			})
 		} else if len(args) == 1 {
-			// Only an output file specified.
+			// Only an output file specified, can only be *stdout*
+			if args[0] != "stdout" {
+				return nil, fmt.Errorf("invalid log file: %s, only %s is allowed", args[0], "stdout")
```
Comment (nchrisdk) [inline plugin/log/setup.go]: Consider shortening the message:
`fmt.Error("Only stdout is allowed")`

**Review by nchrisdk — APPROVED** [review]
Looks good.
Thanks for the quick response!

**Comment by miekg** [comment]
We could remove 'stdout' in is entirety? But that's no backwards compatible, so... meh.

**Comment by stp-ip** [comment]
We haven't released v1.0.0 so we could do a breaking change for v1.0.0.
There won't be a better time for such a breaking change in my view.

**Comment by miekg** [comment]
Good point. @nchrisdk you want to take care of that?
(I'll merge this when tests are done)

**Pull request merged**